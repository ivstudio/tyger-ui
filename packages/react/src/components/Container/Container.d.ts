import type { TBreakpointKey } from '@tyger-ui/system';

import type { BoxProps } from '../Box/Box.d';

export interface ContainerProps extends BoxProps {
    maxWidth?: TBreakpointKey;
    className?: string;
}
