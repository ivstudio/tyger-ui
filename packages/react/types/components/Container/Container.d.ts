import { TBreakpointKey } from '../../styles/mediaQueries';
import { IBox } from '../Box';
interface IContainer extends IBox {
    maxWidth?: TBreakpointKey;
}
declare const Container: ({ children, maxWidth, tag, ...props }: IContainer) => JSX.Element;
export default Container;
