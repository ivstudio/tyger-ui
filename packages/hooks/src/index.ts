export { default as useClickOutside } from './useClickOutside';
export { default as useEventListener } from './useEventListener';
export { default as useIsomorphicLayoutEffect } from './useIsomorphicLayoutEffect';
