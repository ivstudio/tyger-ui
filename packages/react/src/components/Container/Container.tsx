import {
    breakpoint,
    device,
    TBreakpointKey,
} from '@tyger-ui/system/src/mediaQueries';
import styled from 'styled-components';

import { BoxRoot, IBox } from '../Box';

interface IContainer extends IBox {
    maxWidth?: TBreakpointKey;
}

const ContainerRoot = styled(BoxRoot)<{
    maxWidth?: TBreakpointKey;
}>`
    width: 100%;
    margin-left: auto;
    margin-right: auto;
    max-width: ${({ maxWidth }) =>
        maxWidth && `${breakpoint[maxWidth]}!important`};

    @media ${device.sm} {
        max-width: ${breakpoint.sm};
    }

    @media ${device.md} {
        max-width: ${breakpoint.md};
    }

    @media ${device.lg} {
        max-width: ${breakpoint.lg};
    }

    @media ${device.xl} {
        max-width: ${breakpoint.xl};
    }
`;

const Container = ({
    children,
    maxWidth,
    tag = 'div',
    ...props
}: IContainer) => {
    return (
        <ContainerRoot as={tag} maxWidth={maxWidth} {...props}>
            {children}
        </ContainerRoot>
    );
};

export default Container;
