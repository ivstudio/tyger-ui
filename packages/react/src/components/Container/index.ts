'use client';
export * from './Container';
export { default } from './Container';
