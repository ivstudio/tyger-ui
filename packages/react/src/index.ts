// Export all components from the components directory
export { default as Accordion } from './components/Accordion';
export { default as AppBar } from './components/AppBar';
export { default as Backdrop } from './components/Backdrop';
export { default as Box } from './components/Box';
export { default as Button } from './components/Button';
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as Chip } from './components/Chip';
export { default as Container } from './components/Container';
export { default as Drawer } from './components/Drawer';
export { default as HideOnScroll } from './components/HideOnScroll';
export { default as IconButton } from './components/IconButton';
export { default as Input } from './components/Input';
export { default as Modal } from './components/Modal';
export { default as ScrollIndicator } from './components/ScrollIndicator';
export { default as Switch } from './components/Switch';
export { default as Textarea } from './components/Textarea';
export { default as Typography } from './components/Typography';

// Export the GlobalStyles component from the system directory
export { default as GlobalStyles } from '@tyger-ui/system/src/GlobalStyles';

// Export utility functions and variables from the system directory
export {
    breakpoint,
    color,
    darkTheme,
    device,
    lightTheme,
    opacityHex,
    spacing,
    zIndex,
} from '@tyger-ui/system/src/index';
