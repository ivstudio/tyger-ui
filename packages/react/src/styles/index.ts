export * from './ellipsis';
export * from './styledMixins';
