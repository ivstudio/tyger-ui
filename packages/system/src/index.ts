import { fontSize, lineHeight } from './font';
import { spacing } from './spacing';
import { zIndex } from './zIndex';
import { color, darkPalette as dark, lightPalette as light } from './colors';
import { DefaultTheme } from 'styled-components';

export interface ICommonTokens {
    spacing: typeof spacing;
    color: typeof color;
    fontSize: typeof fontSize;
    lineHeight: typeof lineHeight;
    zIndex: typeof zIndex;
}
const commonTokens: ICommonTokens = {
    color,
    spacing,
    fontSize,
    lineHeight,
    zIndex,
};

export const lightPalette = {
    ...commonTokens,
    ...light,
};

export const darkPalette = {
    ...commonTokens,
    ...dark,
};

export const lightTheme: DefaultTheme = lightPalette;
export const darkTheme: DefaultTheme = darkPalette;
export type TTheme = typeof lightTheme | typeof darkTheme;
export type ThemeProps = { theme?: TTheme };
