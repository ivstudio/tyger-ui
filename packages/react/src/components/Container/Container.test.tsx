import { breakpoint } from '@tyger-ui/system/src/mediaQueries';
import React from 'react';

import { lgTextFixture } from '../../../test/fixtures';
import { render } from '../../../test/utils/mockProvider';
import Container from './Container';

describe('Container', () => {
    it('renders Container correctly', () => {
        const { container } = render(<Container>{lgTextFixture}</Container>);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders correct element tag', () => {
        const { container, rerender, debug } = render(
            <Container>{lgTextFixture}</Container>
        );

        expect(container.querySelector('div')).toBeInTheDocument();

        // rerender(<Container tag="section">{lgTextFixture}</Container>);
        // expect(container.querySelector('section')).toBeInTheDocument();

        // rerender(<Container tag="span">{lgTextFixture}</Container>);
        // expect(container.querySelector('span')).toBeInTheDocument();
    });

    // it('has correct width', () => {
    //     const { container, rerender } = render(
    //         <Container maxWidth="sm">{lgTextFixture}</Container>
    //     );

    //     expect(container.firstElementChild).toHaveStyle(
    //         `max-width: ${breakpoint.sm}`
    //     );

    //     rerender(<Container maxWidth="lg">{lgTextFixture}</Container>);
    //     expect(container.firstElementChild).toHaveStyle(
    //         `max-width: ${breakpoint.lg}`
    //     );
    // });
});
