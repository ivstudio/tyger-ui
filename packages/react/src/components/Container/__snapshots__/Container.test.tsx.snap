// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders Container correctly 1`] = `
<div
  class="sc-aXZVg sc-gEvEer jsAQMJ dYYHUr"
>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. 
</div>
`;
